### PR TITLE
Block Syncing enabled for Polygon in Bitcore-Node

### PR DESCRIPTION
**Pull request by escottalexander**
This makes it possible to do block syncing for Polygon. It also contains a new EVM base class that both ETH and MATIC are using. As such there are a lot of files changed but mostly just changes to class names.

### CODE REVIEW

**Review by kajoseph — CHANGES_REQUESTED** [review]

File: packages/bitcore-node/src/providers/chain-state/evm/types.ts [hunk 1]
```diff
@@ -93,17 +93,17 @@ export interface GethTraceCall {
 export type AnyBlock = GethBlock | ErigonBlock;
 export type AnyTransaction = GethTraceTransaction | ErigonTransaction;
 
-export type Networks = 'mainnet' | 'ropsten' | 'rinkeby' | 'goerli' | 'kovan' | 'sepolia';
+export type Networks = 'mainnet' | 'mumbai';
```
[the remark this answers is not in the document]
Comment (escottalexander) [reply packages/bitcore-node/src/providers/chain-state/evm/types.ts]: My bad, think I accidentally used the MATIC types file. Corrected to include networks on both MATIC and ETH.

File: packages/bitcore-node/src/providers/chain-state/evm/types.ts [hunk 2]
```diff
@@ -224,5 +224,17 @@ export interface EthTransactionJSON {
   decodedData?: IAbiDecodedData;
   data: string;
   internal: Array<DecodedTrace>;
-  receipt?: IEthTransaction['receipt'];
+  receipt?: IEVMTransaction['receipt'];
+}
+
+export interface EventLog<T> {
```
[the remark this answers is not in the document]
Comment (escottalexander) [reply packages/bitcore-node/src/providers/chain-state/evm/types.ts:230]: I removed the csp definition and just made sure everything was included in types file.

File: packages/bitcore-node/src/providers/chain-state/evm/models/transaction.ts [hunk 3]
```diff
@@ -206,12 +206,12 @@ export class EthTransactionModel extends BaseTransaction<IEthTransaction> {
...
       return Promise.all(
-        params.txs.map(async (tx: IEthTransaction) => {
+        params.txs.map(async (tx: IEVMTransaction) => {
           const { to, txid, from } = tx;
           const tos = [to];
           const froms = [from];
-
-          const { web3 } = await ETH.getWeb3(network);
+          const CSP = new BaseEVMStateProvider(chain);
+          const { web3 } = await CSP.getWeb3(network);
```
[the remark this answers is not in the document]
Comment (escottalexander) [reply packages/bitcore-node/src/providers/chain-state/evm/models/transaction.ts]: Good call. Moved outside the map function.

File: packages/bitcore-node/src/modules/ethereum/api/csp.ts [hunk 4]
```diff
@@ -1,585 +1,5 @@
...
-  async getCoinsForTx() {
-    return {
-      inputs: [],
-      outputs: []
-    };
-  }
-}
-
-export const ETH = new ETHStateProvider();
+export const ETH = new ETHStateProvider('ETH');
```
[the remark this answers is not in the document]
Comment (escottalexander) [reply packages/bitcore-node/src/modules/ethereum/api/csp.ts]: 👍

**Review by kajoseph — APPROVED** [review]
LGTM. Tested

**Pull request merged**